haystack/nodes/search_engine/web.py

```diff
@@ -0,0 +1,133 @@
+import pydoc
+from typing import List, Dict, Any, Optional, Union, Tuple, Type, Callable
+
+from haystack import BaseComponent, MultiLabel, Document
+from haystack.nodes import PromptNode, PromptTemplate, TopPSampler
+from haystack.nodes.search_engine.base import SearchEngine
+
+
+class WebSearch(BaseComponent):
+    """
+    WebSearch queries a search engine and retrieve results. It can use various search engine providers, e.g. SerperDev,
+    SerpAPI, etc.
+    """
+
+    outgoing_edges = 1
+
+    def __init__(self, api_key: str, search_engine_provider: Union[str, SearchEngine] = "SerperDev", **kwargs):
+        """
+        :param api_key: API key for the search engine provider
+        :param search_engine_provider: Name of the search engine provider, e.g. "SerperDev", "SerpAPI"
+        :param kwargs: Additional parameters to pass to the search engine provider
+        """
+        super().__init__()
+        if isinstance(search_engine_provider, str):
```

Comment:
Let's discuss this part of the code in a call. I wonder whether we can simplify it but it wouldn't block this PR because of it.

Reply:
It can be confusing for users here that we have a variable search_engine_provider that can be of type search_engine but is not the actual search engine we use.

Reply:
Yeah @ZanSara showed me another approach how to solve this with the subclass init method. We can use that approach.